commit 1a8f4a490871df59d1f3cfe28fae4458d2cbef7f
Author: Bart Van Assche <bart.vanassche@sandisk.com>
Date:   Tue Dec 15 16:38:22 2015 +0100

    brd: Fix discard request processing
    
    commit 5e4298be45e83ecdffaabb370eea9396889b07f1 upstream.
    
    Avoid that discard requests with size => PAGE_SIZE fail with
    -EIO. Refuse discard requests if the discard size is not a
    multiple of the page size.
    
    Fixes: 2dbe54957636 ("brd: Refuse improperly aligned discard requests")
    Signed-off-by: Bart Van Assche <bart.vanassche@sandisk.com>
    Reviewed-by: Jan Kara <jack@suse.com>
    Cc: Christoph Hellwig <hch@lst.de>
    Cc: Robert Elliot <elliott@hp.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/block/brd.c b/drivers/block/brd.c
index a5880f4..1914c63 100644
--- a/drivers/block/brd.c
+++ b/drivers/block/brd.c
@@ -338,7 +338,7 @@ static blk_qc_t brd_make_request(struct request_queue *q, struct bio *bio)
 
 	if (unlikely(bio->bi_rw & REQ_DISCARD)) {
 		if (sector & ((PAGE_SIZE >> SECTOR_SHIFT) - 1) ||
-		    bio->bi_iter.bi_size & PAGE_MASK)
+		    bio->bi_iter.bi_size & ~PAGE_MASK)
 			goto io_error;
 		discard_from_brd(brd, sector, bio->bi_iter.bi_size);
 		goto out;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1358_1a8f4a490871df59d1f3cfe28fae4458d2cbef7f.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1358_1a8f4a490871df59d1f3cfe28fae4458d2cbef7f.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
